src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -84,9 +84,8 @@ public GenericWebHostBuilder(IHostBuilder builder, WebHostBuilderOptions options
 
                 // REVIEW: This is bad since we don't own this type. Anybody could add one of these and it would mess things up
                 // We need to flow this differently
-                var listener = new DiagnosticListener("Microsoft.AspNetCore");
-                services.TryAddSingleton<DiagnosticListener>(listener);
-                services.TryAddSingleton<DiagnosticSource>(listener);
+                services.TryAddSingleton(sp => new DiagnosticListener("Microsoft.AspNetCore"));
```

Comment:
I guess this could have been:

```C#
var listener = new DiagnosticListener("Microsoft.AspNetCore");
services.TryAddSingleton<DiagnosticListener>(_ => listener);
services.TryAddSingleton<DiagnosticSource>(listener);
```